### PR TITLE
refactor: rename txHashList to txHash

### PR DESCRIPTION
**Pull request by mohitsethia**
# Description
Issue [link](https://github.com/zeta-chain/node/issues/1995)
In the proto file for outboundtracker:

```
message OutBoundTracker {
  string index = 1; // format: "chain-nonce"
  int64 chain_id = 2;
  uint64 nonce = 3;
  repeated TxHashList hash_list = 4;
}
```
It doesn't make sense to call TxHashList since this is the element of the list. Should be renamed TxHash

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **Refactor**
  - Renamed `TxHashList` to `TxHash` to improve clarity and consistency across the application.
  - Updated references to `hashList` within the `OutboundTracker` to reflect the new naming convention.

### CODE REVIEW

File: proto/zetachain/zetacore/crosschain/outbound_tracker.proto
Comment (kingpinXD) [inline proto/zetachain/zetacore/crosschain/outbound_tracker.proto:15]: This change might need a migration script to manage migrating any existing outbound trackers to the new format when the upgrade happens

Comment (mohitsethia) [reply proto/zetachain/zetacore/crosschain/outbound_tracker.proto:15]: Thanks for the comment!
Can you please share if there is any reference that I can use? Or maybe explain a bit more so that I can implement that on my own.

Comment (lumtis) [reply proto/zetachain/zetacore/crosschain/outbound_tracker.proto:15]: You don't need migration script here if you don't change data types

**Review by lumtis — COMMENTED** [review]
@mohitsethia can you run `make generate` to run file generation given the changes?

**Comment by mohitsethia** [comment]
@lumtis I need to change the same variable name in all the places I think. And running `make generate` only removed most of the ts files, but I only commited the changes that were made to outbound_tracker.pb.go file.
I will fix it by EOD.

**Comment by lumtis** [comment]
> @lumtis I need to change the same variable name in all the places I think. And running `make generate` only removed most of the ts files, but I only commited the changes that were made to outbound_tracker.pb.go file. I will fix it by EOD.

Yeah, everything changes by `make generate` needs to be committed, as the CI run the same command and check if there is a difference in the diff

**Comment by mohitsethia** [comment]
There are 441 files deleted because of it, is this expected? 😓

**Comment by lumtis** [comment]
> There are 441 files deleted because of it, is this expected? 😓

Now, I think there is an issue when you run `make generate`

Can you show the logs printed by the command? Thanks

**Comment by mohitsethia** [comment]
It was some issue with file not found, but now it resolved by itself.
```
➜ make generate
--> Formatting Protobuf files
--> Removing old Go types 
--> Generating Protobuf files
Generating gogo proto code
--> Generating OpenAPI specs
rm: ./docs/openapi/zetachain.swagger.yaml: No such file or directory
--> Generating module documentation
--> Generating TypeScript bindings
./scripts/protoc-gen-typescript.sh: line 9: typescript/package.json: No such file or directory
--> Generating zetacored documentation
--> Generating mocks
Mocks generated.
Fixing long lines...
Formatting code...
Code is formatted
```

**Review by lumtis — APPROVED** [review]

**Comment by mohitsethia** [comment]
I need one more approval before merging this PR. @kingpinXD can you please check this too? Thanks!

**Review by fbac — APPROVED** [review]

**Pull request merged**
